src/containers/Tag/TagList/TagList.tsx

```diff
@@ -44,66 +95,147 @@ export const TagList: React.SFC<TagListProps> = (props) => {
+  if (countData) {
+    tagCount = countData.countTags;
   }
 
   return (
-    <div>
-      <div className={styles.AddButtton}>
-        <Button variant="contained" color="primary" onClick={() => setNewTag(true)}>
-          New Tag
-        </Button>
+    <>
+      <div className={styles.Header}>
+        <Typography variant="h5" className={styles.Title}>
+          Tags
+        </Typography>
+        <div className={styles.Buttons}>
+          <IconButton className={styles.IconButton} onClick={() => setSearchOpen(!searchOpen)}>
+            <SearchIcon className={styles.SearchIcon}></SearchIcon>
+          </IconButton>
+          <form onSubmit={handleSearch}>
+            <div className={searchOpen ? styles.SearchBar : styles.HideSearchBar}>
+              <InputBase
+                defaultValue={searchVal}
+                className={searchOpen ? styles.ShowSearch : styles.HideSearch}
+                name="nameSearch"
+              />
+              {searchOpen ? (
+                <div className={styles.ResetSearch} onClick={() => setSearchVal('')}>
+                  <Divider orientation="vertical" />
+                  <CloseIcon className={styles.CloseIcon}></CloseIcon>
+                </div>
+              ) : null}
+            </div>
+          </form>
+
+          {/* Not sure how to override Material UI button styling without too much hassle */}
+          <Button
+            style={{
+              height: '80%',
+              textTransform: 'capitalize',
+              borderRadius: '20px',
+              marginLeft: '10px',
+            }}
+            color="primary"
+            variant="contained"
+            disableElevation={true}
+            onClick={() => setNewTag(true)}
+          >
+            Add New
+          </Button>
+        </div>
       </div>
-      <br />
-      <br />
-      <TableContainer component={Paper}>
-        <Table className={styles.Table} aria-label="tag listing">
-          <TableHead>
-            <TableRow>
-              <TableCell>Name</TableCell>
-              <TableCell>Description</TableCell>
-              <TableCell></TableCell>
-            </TableRow>
-          </TableHead>
-          <TableBody>{listing}</TableBody>
-        </Table>
-      </TableContainer>
-    </div>
+
+      {/* Table */}
+      {tagList ? (
+        <DataTable
+          className={styles.Table}
+          columns={columns}
+          data={tagList}
+          pagination
+          noHeader
+          onChangePage={(page, totalRows) => setPageNum(page)} // passes (page, totalRows)
+          onChangeRowsPerPage={(currentRowsPerPage, currentPage) => setPageRows(currentRowsPerPage)} // passes (currentRowsPerPage, currentPage)
+          paginationTotalRows={tagCount}
+          paginationServer
+          paginationPerPage={pageRows}
+          sortServer
+          // BUG: When trying to sort on a column, you have to click the column twice in order to see any changes.
```

Comment:
Very weird bug about sorting columns, may be related to the React-Data-Table package instead of my current functionality.

Reply:
Ok. I would suggest you post this issue in the "react-data-table" issue queue and see if they have any workaround for the same.